woocommerce/woocommerce-ios
# Set payment method and payment method title for Cash Orders

Closes: #12347
More context: pdfdoF-5ai-p2


## Description

When collecting cash payments through the app, the order status is marked as `complete` without setting the `payment_method` and `payment_method_title` as its done on the web.

Cash payment logic is initiated by `PaymentMethodViewModel` through `markOrderAsPaidByCash` method. At the moment, all it does to the order is update its status to completed.

### Solution

1. Use `updateOrder` instead of `updateOrderStatus` action that allows us to set modified `Order` object, and pass an array of fields that should be updated on the backend.
2. Use `PaymentGateway.Constants.cashOnDeliveryGatewayID` (`cod`) as `paymentMethodID`
3. Fetch Cash on Delivery (COD) payment gateway through `viewStorage.loadPaymentGateway`. If it doesn't exist, set a default "Pay in Person" title

## Testing information

1. Create an order in the app
6. Open detail of the order
7. Tap on 'Collect Payment'
8. Select Cash
9. Confirm
10. Confirm the order is marked as completed, and payment_method and payment_method_title are set. Payment method title can be visible in order details or receipts.

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/4062343/1a4e01df-c8ae-4a53-93ab-1580429a36e5

---
